### PR TITLE
feat(studyspecificdataset)/implement file upload

### LINKED ISSUE #360: Implement a StudyLogoService and make IAzureBlobStorageService general
Right now, logo crud is handled by IAzureBlobStorageService. The name of this service is to general compared to the job it performs. IAzureBlobStorageService should be a general service that handles blob storage CRUD. There should be a IStudyLogoService that handles only Study logo CRUD, and it can use IAzureBlobStorageService to perform the job.

### LINKED ISSUE #442: StudySpecificDataset: Implement (multiple) file upload
Create resource group
Creat Storage account
Upload files

### PR DESCRIPTION
**Pull request by krsteen**
Closes #360 
Closes #442

### CODE REVIEW

File: src/Sepes.Infrastructure/Constants/Auth/AzureRoleDefinitionId.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Sepes.Infrastructure.Constants.Auth
+{
+    public static class AzureRoleDefinitionId
+    {
+        public const string READ = "acdd72a7-3385-48ef-bd42-f606fba81ae7";
```
Comment (HansKristianVikOlsen) [inline src/Sepes.Infrastructure/Constants/Auth/AzureRoleDefinitionId.cs:5]: What is this?

File: src/Sepes.Infrastructure/Service/Azure/AzureApiServiceBase.cs [hunk 2]
```diff
@@ -95,9 +108,30 @@ protected async Task<T> GetResponse<T>(string url, bool needsAuth = true, Cancel
                     if (needsAuth)
                     {
                         apiRequestClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
-                    }                   
+                    }
```
Comment (HansKristianVikOlsen) [inline src/Sepes.Infrastructure/Service/Azure/AzureApiServiceBase.cs:111]: needsAuth is always true? Is there a point with and IF on this?

**Review by HansKristianVikOlsen — APPROVED** [review]
Looks good!

File: src/Sepes.Infrastructure/Constants/Auth/AzureRoleDefinitionId.cs (hunk 1 above)
Comment (krsteen) [reply src/Sepes.Infrastructure/Constants/Auth/AzureRoleDefinitionId.cs:5]: Ah, it's for giving the user that created the data source access to the storage account in Az. That is the id for the generic role "Read" in Azure

File: src/Sepes.Infrastructure/Service/Azure/AzureApiServiceBase.cs (hunk 2 above)
Comment (krsteen) [reply src/Sepes.Infrastructure/Service/Azure/AzureApiServiceBase.cs:111]: Somebody also sends false as an argument, like AzureCostManagementService which implements that class

**Pull request merged**